src/app/ClusterStateCache.cpp

```diff
@@ -44,20 +42,15 @@ CHIP_ERROR ClusterStateCache::UpdateCache(const ConcreteDataAttributePath & aPat
 
     if (apData)
     {
-        handle = System::PacketBufferHandle::New(chip::app::kMaxSecureSduLengthBytes);
-
-        writer.Init(std::move(handle), false);
-
+        Platform::ScopedMemoryBuffer<uint8_t> backingBuffer;
+        uint32_t totalBufSize = apData->GetTotalLength();
```

Comment:
But that assumes that the reader is positioned at the start of its buffer and the element ends at the end of the buffer, at least if we are to avoid over-allocating.  This _really_  matters for read responses with multiple attribute values in them.

Could we not just compute the size of the element directly and allocate the buffer to the right size?

Reply:
from the current code path, OnAttributeData takes concrete path, and then pass the data to this UpdateCache, so this tlvreader would have only one element inside, there is no chance for multiple attribute?

Reply:
agree, in updated PR, I have created GetElementTLVSize utility function and got the actual element size via using tlvwriter and CopyElement and GetLengthWritten